In [1]:
#Libraries basics

import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import numpy as np # linear algebra

In [2]:
#Libraries by visualization

import plotly.express as px
import matplotlib.pyplot as plt

The goal of the analysis is to see the:

* The relationship between the Publisher and Total Sales
* The relationship between Developer and Total Sales
* The relationship between Sales and Platform
* The relationship between Total Sales and User/Critic Score

# 1. Ask

Existen muchas preguntas que podemos formular de acuerdo a este conjunto de datos, debido a que tiene muchas variables de las mas principales y en las que nos centraremos son:

Que empresa tiene el mejor rendimiento en venta de videojuegos?

Cual fue el ano con mejor rendimiento y en que continente?

Cual es el genero con mejores ventas?

Los puntajes determinan el exito de un videojuego?

Estas son una de las preguntas que pueden responderse, sin embargo se pueden responder y analizar otras.

There are many questions that we can ask according to this data set, because it has many variables, the most important of which we will focus on are the following:

* Which company has the best performance in video game sales?

* Which was the year with the best performance and in which continent?

* What is the genre with the best sales?

* Do the scores determine the success of a video game?

These are some of the questions that can be answered, but others can be answered and analyzed.

# 2. Prepare

This data is freely licensed through Kaggle, from this publication.

License: CC0 1.0 Universal (CC0 1.0) - Public Domain Dedication
No Copyright - You can copy, modify, distribute and perform the work, even for commercial purposes, all without asking permission. See Other Information.

[Link](https://www.kaggle.com/datasets/thedevastator/global-video-game-sales-ratings?datasetId=2885428)

We have 3 types of data sets, including clean and tage data, for this notebook we will use the dirty or unclean data in order to make use of basic of Python concepts and concepts learned during the Google Data Analysis course.

This notebook has executable code in R for more information you can search for the statement or Python tutorials.

[Python Concepts](https://www.w3schools.com/python/python_getstarted.asp)

In [3]:
data = pd.read_csv('./Data/Raw_Data.csv')
data.head()

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1           NaN        NaN         NaN       NaN    NaN  
2          73.0        8.3       709.0  Nintendo      E  
3          73.0          8       192.0  Nintendo      E  
4           NaN        NaN         NaN       NaN    NaN

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       10015 non-null  object 
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(9), object(7)
memory usage: 2.0+ MB


**Note1: "Year_of_Release" should be integer instead of float**

**Note2: "User_Score" should be float instead of object**

### 2.1 Categorical/Object Data

In [5]:
object_type = data.dtypes == 'object'
filtered_list = object_type.index[object_type].tolist()

data[filtered_list].nunique()

Name          11562
Platform         31
Genre            12
Publisher       581
User_Score       96
Developer      1696
Rating            8
dtype: int64

**Note: 'Genre' and 'Rating' columns could be casted to category type**

### 2.2 Numerical Data

In [6]:
data.describe()

Year_of_Release      NA_Sales      EU_Sales      JP_Sales  \
count     16450.000000  16719.000000  16719.000000  16719.000000   
mean       2006.487356      0.263330      0.145025      0.077602   
std           5.878995      0.813514      0.503283      0.308818   
min        1980.000000      0.000000      0.000000      0.000000   
25%        2003.000000      0.000000      0.000000      0.000000   
50%        2007.000000      0.080000      0.020000      0.000000   
75%        2010.000000      0.240000      0.110000      0.040000   
max        2020.000000     41.360000     28.960000     10.220000   

        Other_Sales  Global_Sales  Critic_Score  Critic_Count    User_Count  
count  16719.000000  16719.000000   8137.000000   8137.000000   7590.000000  
mean       0.047332      0.533543     68.967679     26.360821    162.229908  
std        0.186710      1.547935     13.938165     18.980495    561.282326  
min        0.000000      0.010000     13.000000      3.000000      4.000000  
25%        0.000000      0.060000     60.000000     12.000000     10.000000  
50%        0.010000      0.170000     71.000000     21.000000     24.000000  
75%        0.030000      0.470000     79.000000     36.000000     81.000000  
max       10.570000     82.530000     98.000000    113.000000  10665.000000

# 3.Process

### 3.1 Missing Values

In [7]:
# Number of missing data

missing_values= data.isna().sum()
missing_values

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

**Note 4: For resolving this problem we have differents alternatives in my case i drop this values in other notebook i used Clean Data Analysis**

In [8]:
# Percentage of data missing

total_cells = np.product(data.shape)
total_missing_values = missing_values.sum()

missing_percent = (total_missing_values/total_cells)*100
print(f'{total_missing_values} out of {total_cells} ({missing_percent:.2f}%) missing')

46716 out of 267504 (17.46%) missing


In [9]:
# Drop NA Values

data = data.dropna(how='any')\
    .reset_index(drop=True)\
    .copy()

In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6825 entries, 0 to 6824
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             6825 non-null   object 
 1   Platform         6825 non-null   object 
 2   Year_of_Release  6825 non-null   float64
 3   Genre            6825 non-null   object 
 4   Publisher        6825 non-null   object 
 5   NA_Sales         6825 non-null   float64
 6   EU_Sales         6825 non-null   float64
 7   JP_Sales         6825 non-null   float64
 8   Other_Sales      6825 non-null   float64
 9   Global_Sales     6825 non-null   float64
 10  Critic_Score     6825 non-null   float64
 11  Critic_Count     6825 non-null   float64
 12  User_Score       6825 non-null   object 
 13  User_Count       6825 non-null   float64
 14  Developer        6825 non-null   object 
 15  Rating           6825 non-null   object 
dtypes: float64(9), object(7)
memory usage: 853.2+ KB


### 3.2 Duplicated values

In [11]:
#This colums has duplicated values

data[data.duplicated()]

Empty DataFrame
Columns: [Name, Platform, Year_of_Release, Genre, Publisher, NA_Sales, EU_Sales, JP_Sales, Other_Sales, Global_Sales, Critic_Score, Critic_Count, User_Score, User_Count, Developer, Rating]
Index: []

In [12]:
data.loc[data.duplicated(subset=['Name'])]

Name Platform  Year_of_Release     Genre  \
13                   Grand Theft Auto V     X360           2013.0    Action   
20           Call of Duty: Black Ops II     X360           2012.0   Shooter   
22       Call of Duty: Modern Warfare 3      PS3           2011.0   Shooter   
26              Call of Duty: Black Ops      PS3           2010.0   Shooter   
27                   Grand Theft Auto V      PS4           2014.0    Action   
...                                 ...      ...              ...       ...   
6812        Saints Row: Gat out of Hell     XOne           2015.0    Action   
6814           Legacy of Kain: Defiance       PC           2003.0    Action   
6817           Sébastien Loeb Rally Evo     XOne           2016.0    Racing   
6821     Mortal Kombat: Deadly Alliance      GBA           2002.0  Fighting   
6822  Metal Gear Solid V: Ground Zeroes       PC           2014.0    Action   

                         Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
13            Take-Two Interactive      9.66      5.14      0.06         1.41   
20                      Activision      8.25      4.24      0.07         1.12   
22                      Activision      5.54      5.73      0.49         1.57   
26                      Activision      5.99      4.37      0.48         1.79   
27            Take-Two Interactive      3.96      6.31      0.38         1.97   
...                            ...       ...       ...       ...          ...   
6812                   Deep Silver      0.00      0.01      0.00         0.00   
6814             Eidos Interactive      0.00      0.01      0.00         0.00   
6817               Milestone S.r.l      0.00      0.01      0.00         0.00   
6821                  Midway Games      0.01      0.00      0.00         0.00   
6822  Konami Digital Entertainment      0.00      0.01      0.00         0.00   

      Global_Sales  Critic_Score  Critic_Count User_Score  User_Count  \
13           16.27          97.0          58.0        8.1      3711.0   
20           13.67          83.0          73.0        4.8      2256.0   
22           13.32          88.0          39.0        3.2      5234.0   
26           12.63          88.0          58.0        6.4      1094.0   
27           12.61          97.0          66.0        8.3      2899.0   
...            ...           ...           ...        ...         ...   
6812          0.01          65.0          16.0          6        27.0   
6814          0.01          70.0          11.0        8.7        67.0   
6817          0.01          63.0           8.0        8.2        22.0   
6821          0.01          81.0          12.0        8.8         9.0   
6822          0.01          80.0          20.0        7.6       412.0   

                              Developer Rating  
13                       Rockstar North      M  
20                             Treyarch      M  
22    Infinity Ward, Sledgehammer Games      M  
26                             Treyarch      M  
27                       Rockstar North      M  
...                                 ...    ...  
6812                      Volition Inc.      M  
6814                    Nixxes Software      M  
6817                    Milestone S.r.l      E  
6821                    Criterion Games      M  
6822                 Kojima Productions      M  

[2448 rows x 16 columns]

**Note:We check the name of videogames is the same but Platform is different**

In [13]:
data.loc[data.duplicated(subset=['Name','Platform'])]

Name Platform  Year_of_Release   Genre  \
1016  Need for Speed: Most Wanted     X360           2005.0  Racing   
5675  Need for Speed: Most Wanted       PC           2012.0  Racing   
6755                Madden NFL 13      PS3           2012.0  Sports   

            Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
1016  Electronic Arts       1.0      0.13      0.02         0.10   
5675  Electronic Arts       0.0      0.06      0.00         0.02   
6755  Electronic Arts       0.0      0.01      0.00         0.00   

      Global_Sales  Critic_Score  Critic_Count User_Score  User_Count  \
1016          1.25          83.0          54.0        8.5       134.0   
5675          0.08          82.0          19.0        8.5       525.0   
6755          0.01          83.0          22.0        5.5       101.0   

       Developer Rating  
1016   EA Canada      T  
5675   Black Box      T  
6755  EA Tiburon      E

**Note: After research, "Need for Speed: Most Wanted" have two versions in 2005 and 2012, therefore, I will rename these values.**

In [14]:
data.loc[1016,'Name'] = 'Need for Speed: Most Wanted 2005'
data.loc[5675,'Name'] = 'Need for Speed: Most Wanted 2012'
data.loc[[1016, 5675],'Name']

1016    Need for Speed: Most Wanted 2005
5675    Need for Speed: Most Wanted 2012
Name: Name, dtype: object

In [15]:
data.loc[data['Name'] == 'Madden NFL 13']

Name Platform  Year_of_Release   Genre        Publisher  \
347   Madden NFL 13     X360           2012.0  Sports  Electronic Arts   
408   Madden NFL 13      PS3           2012.0  Sports  Electronic Arts   
3366  Madden NFL 13      PSV           2012.0  Sports  Electronic Arts   
3879  Madden NFL 13     WiiU           2012.0  Sports  Electronic Arts   
6755  Madden NFL 13      PS3           2012.0  Sports  Electronic Arts   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
347       2.53      0.15       0.0         0.17          2.86          81.0   
408       2.11      0.22       0.0         0.23          2.56          83.0   
3366      0.28      0.00       0.0         0.02          0.30          63.0   
3879      0.21      0.00       0.0         0.02          0.23          75.0   
6755      0.00      0.01       0.0         0.00          0.01          83.0   

      Critic_Count User_Score  User_Count   Developer Rating  
347           36.0        5.8       179.0  EA Tiburon      E  
408           22.0        5.5       101.0  EA Tiburon      E  
3366           6.0        7.3        38.0  EA Tiburon      E  
3879           9.0        6.7        30.0  EA Tiburon      E  
6755          22.0        5.5       101.0  EA Tiburon      E

**Note: Row 408 and row 6755 are same game and on same platform, confirm they're duplicated but index 6755 present errors**

In [16]:
# Drop row 6755

data = data.loc[~data.duplicated(subset=['Name','Platform'])].reset_index(drop=True).copy()

In [17]:
data.loc[data.duplicated(subset=['Name','Platform'])]

Empty DataFrame
Columns: [Name, Platform, Year_of_Release, Genre, Publisher, NA_Sales, EU_Sales, JP_Sales, Other_Sales, Global_Sales, Critic_Score, Critic_Count, User_Score, User_Count, Developer, Rating]
Index: []

### 3.3 Inconsistent Data

In [18]:
check1 = data.groupby(['Name','Platform','Year_of_Release','Genre','Publisher','Developer','Rating'])['Name'].count().to_frame()

check1.query('Name > 1')

Empty DataFrame
Columns: [Name]
Index: []

In [19]:
check2 = data.groupby(['Name', 'NA_Sales', 'EU_Sales', 'JP_Sales',
                            'Other_Sales', 'Global_Sales', 'Critic_Score',
                            'Critic_Count', 'User_Score', 'User_Count'])['Name'].count().to_frame()
check2.query('Name > 1')

Empty DataFrame
Columns: [Name]
Index: []

**Given the result of "consis_check1" and "consis_check2", there's no inconsistent data entry for every game.**

### 3.4 Data Formating

In [20]:
#Preview View of data
data.head()

Name Platform  Year_of_Release     Genre Publisher  \
0             Wii Sports      Wii           2006.0    Sports  Nintendo   
1         Mario Kart Wii      Wii           2008.0    Racing  Nintendo   
2      Wii Sports Resort      Wii           2009.0    Sports  Nintendo   
3  New Super Mario Bros.       DS           2006.0  Platform  Nintendo   
4               Wii Play      Wii           2006.0      Misc  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     15.68     12.76      3.79         3.29         35.52          82.0   
2     15.61     10.93      3.28         2.95         32.77          80.0   
3     11.28      9.14      6.50         2.88         29.80          89.0   
4     13.96      9.18      2.93         2.84         28.92          58.0   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1          73.0        8.3       709.0  Nintendo      E  
2          73.0          8       192.0  Nintendo      E  
3          65.0        8.5       431.0  Nintendo      E  
4          41.0        6.6       129.0  Nintendo      E

In [21]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6824 entries, 0 to 6823
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             6824 non-null   object 
 1   Platform         6824 non-null   object 
 2   Year_of_Release  6824 non-null   float64
 3   Genre            6824 non-null   object 
 4   Publisher        6824 non-null   object 
 5   NA_Sales         6824 non-null   float64
 6   EU_Sales         6824 non-null   float64
 7   JP_Sales         6824 non-null   float64
 8   Other_Sales      6824 non-null   float64
 9   Global_Sales     6824 non-null   float64
 10  Critic_Score     6824 non-null   float64
 11  Critic_Count     6824 non-null   float64
 12  User_Score       6824 non-null   object 
 13  User_Count       6824 non-null   float64
 14  Developer        6824 non-null   object 
 15  Rating           6824 non-null   object 
dtypes: float64(9), object(7)
memory usage: 853.1+ KB


In [22]:
data['Year_of_Release'] = data['Year_of_Release'].astype('int')
data['User_Score'] = data['User_Score'].astype('float')
data[['Genre','Rating']] = data[['Genre','Rating']].astype('category')

data.dtypes

Name                 object
Platform             object
Year_of_Release       int32
Genre              category
Publisher            object
NA_Sales            float64
EU_Sales            float64
JP_Sales            float64
Other_Sales         float64
Global_Sales        float64
Critic_Score        float64
Critic_Count        float64
User_Score          float64
User_Count          float64
Developer            object
Rating             category
dtype: object

# 4.Analyze



In this moment for answer the initial questions

Que empresa tiene el mejor rendimiento en venta de videojuegos?

Cual fue el ano con mejor rendimiento y en que continente?

Cual es el genero con mejores ventas?

Los puntajes determinan el exito de un videojuego?

* Which company has the best performance in video game sales?

* Which was the year with the best performance and in which continent?

* What is the genre with the best sales?

* Do the scores determine the success of a video game?



Ordenación de datos en hojas de cálculo y mediante la escritura de consultas SQL

Filtrado de datos en hojas de cálculo y mediante la escritura de consultas SQL

Convertir datos

Formatear datos

Corroborar los procesos de análisis de datos

Obtener comentarios y apoyo de terceros durante el análisis de datos


In [23]:
data.head()

Name Platform  Year_of_Release     Genre Publisher  \
0             Wii Sports      Wii             2006    Sports  Nintendo   
1         Mario Kart Wii      Wii             2008    Racing  Nintendo   
2      Wii Sports Resort      Wii             2009    Sports  Nintendo   
3  New Super Mario Bros.       DS             2006  Platform  Nintendo   
4               Wii Play      Wii             2006      Misc  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     15.68     12.76      3.79         3.29         35.52          82.0   
2     15.61     10.93      3.28         2.95         32.77          80.0   
3     11.28      9.14      6.50         2.88         29.80          89.0   
4     13.96      9.18      2.93         2.84         28.92          58.0   

   Critic_Count  User_Score  User_Count Developer Rating  
0          51.0         8.0       322.0  Nintendo      E  
1          73.0         8.3       709.0  Nintendo      E  
2          73.0         8.0       192.0  Nintendo      E  
3          65.0         8.5       431.0  Nintendo      E  
4          41.0         6.6       129.0  Nintendo      E

### 4.1 The best company in global_sales

In [24]:
#Top10 of Videogames with more global sales
data.sort_values(by=['Global_Sales'],ascending=False).head(10)


Name Platform  Year_of_Release     Genre  \
0                 Wii Sports      Wii             2006    Sports   
1             Mario Kart Wii      Wii             2008    Racing   
2          Wii Sports Resort      Wii             2009    Sports   
3      New Super Mario Bros.       DS             2006  Platform   
4                   Wii Play      Wii             2006      Misc   
5  New Super Mario Bros. Wii      Wii             2009  Platform   
6              Mario Kart DS       DS             2005    Racing   
7                    Wii Fit      Wii             2007    Sports   
8         Kinect Adventures!     X360             2010      Misc   
9               Wii Fit Plus      Wii             2009    Sports   

                Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0                Nintendo     41.36     28.96      3.77         8.45   
1                Nintendo     15.68     12.76      3.79         3.29   
2                Nintendo     15.61     10.93      3.28         2.95   
3                Nintendo     11.28      9.14      6.50         2.88   
4                Nintendo     13.96      9.18      2.93         2.84   
5                Nintendo     14.44      6.94      4.70         2.24   
6                Nintendo      9.71      7.47      4.13         1.90   
7                Nintendo      8.92      8.03      3.60         2.15   
8  Microsoft Game Studios     15.00      4.89      0.24         1.69   
9                Nintendo      9.01      8.49      2.53         1.77   

   Global_Sales  Critic_Score  Critic_Count  User_Score  User_Count  \
0         82.53          76.0          51.0         8.0       322.0   
1         35.52          82.0          73.0         8.3       709.0   
2         32.77          80.0          73.0         8.0       192.0   
3         29.80          89.0          65.0         8.5       431.0   
4         28.92          58.0          41.0         6.6       129.0   
5         28.32          87.0          80.0         8.4       594.0   
6         23.21          91.0          64.0         8.6       464.0   
7         22.70          80.0          63.0         7.7       146.0   
8         21.81          61.0          45.0         6.3       106.0   
9         21.79          80.0          33.0         7.4        52.0   

             Developer Rating  
0             Nintendo      E  
1             Nintendo      E  
2             Nintendo      E  
3             Nintendo      E  
4             Nintendo      E  
5             Nintendo      E  
6             Nintendo      E  
7             Nintendo      E  
8  Good Science Studio      E  
9             Nintendo      E

In [25]:
#Considering that in the list the most 10 publisher data
data['Publisher'].value_counts().head(10)

Electronic Arts                 943
Ubisoft                         496
Activision                      492
Sony Computer Entertainment     316
THQ                             307
Nintendo                        291
Sega                            284
Take-Two Interactive            273
Konami Digital Entertainment    244
Namco Bandai Games              239
Name: Publisher, dtype: int64

In [26]:
global_sales=data.groupby(['Publisher','Global_Sales']).agg('sum').sort_values(by='Global_Sales',ascending=False)
global_sales.head(10)

C:\Users\Jorge Hdz\AppData\Local\Temp\ipykernel_12636\4050219238.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  global_sales=data.groupby(['Publisher','Global_Sales']).agg('sum').sort_values(by='Global_Sales',ascending=False)


Year_of_Release  NA_Sales  EU_Sales  \
Publisher              Global_Sales                                        
Nintendo               82.53                    2006     41.36     28.96   
                       35.52                    2008     15.68     12.76   
                       32.77                    2009     15.61     10.93   
                       29.80                    2006     11.28      9.14   
                       28.92                    2006     13.96      9.18   
                       28.32                    2009     14.44      6.94   
                       23.21                    2005      9.71      7.47   
                       22.70                    2007      8.92      8.03   
Microsoft Game Studios 21.81                    2010     15.00      4.89   
Nintendo               21.79                    2009      9.01      8.49   

                                     JP_Sales  Other_Sales  Critic_Score  \
Publisher              Global_Sales                                        
Nintendo               82.53             3.77         8.45          76.0   
                       35.52             3.79         3.29          82.0   
                       32.77             3.28         2.95          80.0   
                       29.80             6.50         2.88          89.0   
                       28.92             2.93         2.84          58.0   
                       28.32             4.70         2.24          87.0   
                       23.21             4.13         1.90          91.0   
                       22.70             3.60         2.15          80.0   
Microsoft Game Studios 21.81             0.24         1.69          61.0   
Nintendo               21.79             2.53         1.77          80.0   

                                     Critic_Count  User_Score  User_Count  
Publisher              Global_Sales                                        
Nintendo               82.53                 51.0         8.0       322.0  
                       35.52                 73.0         8.3       709.0  
                       32.77                 73.0         8.0       192.0  
                       29.80                 65.0         8.5       431.0  
                       28.92                 41.0         6.6       129.0  
                       28.32                 80.0         8.4       594.0  
                       23.21                 64.0         8.6       464.0  
                       22.70                 63.0         7.7       146.0  
Microsoft Game Studios 21.81                 45.0         6.3       106.0  
Nintendo               21.79                 33.0         7.4        52.0

**We see that Nintendo is the most important company in global sales in differents years**

### 4.2 Better Year in Sales that differents regions

In [27]:
#Global Sales in differents Years top10
years_g = (data.groupby(['Year_of_Release'])['Global_Sales'].agg(['mean','sum']))
years_g = years_g.sort_values(by='sum',ascending=False)
years_g.head(10)


mean     sum
Year_of_Release                  
2008             0.826216  489.12
2009             0.836091  459.85
2007             0.773271  456.23
2006             0.789242  416.72
2010             0.962611  412.96
2011             0.846998  383.69
2005             0.594875  334.32
2004             0.676008  321.78
2012             0.935641  291.92
2002             0.634813  288.84

**The games in 2008 release have the most global_sales cantity sum**

In [31]:
#NorteAmerica Sales
years_na = (data.groupby(['Year_of_Release'])['NA_Sales'].agg(['mean','sum']))
years_na = years_na.sort_values(by='sum',ascending=False)
years_na.head(10)

mean     sum
Year_of_Release                  
2008             0.432855  256.25
2007             0.399339  235.61
2009             0.421309  231.72
2006             0.427443  225.69
2010             0.497063  213.24
2011             0.420795  190.62
2005             0.316993  178.15
2004             0.365294  173.88
2002             0.359912  163.76
2003             0.287309  143.08

**2008 is the most important year by NorteAmerican Sales that the same form of the Global**

In [32]:
#Europe Sales in differents Years top10
years_eu = (data.groupby(['Year_of_Release'])['EU_Sales'].agg(['mean','sum']))
years_eu = years_eu.sort_values(by='sum',ascending=False)
years_eu.head(10)

mean     sum
Year_of_Release                  
2009             0.261018  143.56
2008             0.231943  137.31
2010             0.303333  130.13
2011             0.282252  127.86
2007             0.211373  124.71
2006             0.197973  104.53
2012             0.317532   99.07
2013             0.359173   95.54
2005             0.154270   86.70
2002             0.184681   84.03

**2009 is the most important year by Europe Sales**

In [33]:
#Japon Sales in differents Years top10
years_jp = (data.groupby(['Year_of_Release'])['JP_Sales'].agg(['mean','sum']))
years_jp = years_jp.sort_values(by='sum',ascending=False)
years_jp.head(10)

mean    sum
Year_of_Release                 
2006             0.076572  40.43
2005             0.068025  38.23
2008             0.063209  37.42
2007             0.059390  35.04
2009             0.062327  34.28
2012             0.087692  27.36
2010             0.058718  25.19
2001             0.092070  23.57
2011             0.051126  23.16
2004             0.047773  22.74

**The games of 2016 genering most sales in Japon**

### 4.3 Genre most popular and Genre with more sales

In [34]:
#The genre with more games or most popular
data['Genre'].value_counts()

Action          1630
Sports           942
Shooter          864
Role-Playing     712
Racing           581
Platform         403
Misc             384
Fighting         378
Simulation       297
Strategy         267
Adventure        248
Puzzle           118
Name: Genre, dtype: int64

In [36]:
#Genres most sales in Global
genre = (data.groupby(['Genre'])['Global_Sales'].agg(['mean','sum']))
genre = genre.sort_values(by='sum',ascending=False)
genre.head(10)



mean      sum
Genre                          
Action        0.738135  1203.16
Sports        0.885180   833.84
Shooter       0.945000   816.48
Role-Playing  0.704171   501.37
Racing        0.819656   476.22
Misc          1.084010   416.26
Platform      0.937469   377.80
Fighting      0.661243   249.95
Simulation    0.682492   202.70
Adventure     0.325605    80.75

In [39]:
#Genre in NA Sales
genre_na = (data.groupby(['Genre'])['NA_Sales'].agg(['mean','sum']))
genre_na = genre_na.sort_values(by='sum',ascending=False)
genre_na.head(10)

mean     sum
Genre                         
Action        0.362718  591.23
Sports        0.485690  457.52
Shooter       0.519398  448.76
Racing        0.388279  225.59
Misc          0.578255  222.05
Role-Playing  0.308329  219.53
Platform      0.480397  193.60
Fighting      0.360820  136.39
Simulation    0.310168   92.12
Adventure     0.156492   38.81

In [40]:
#Genre in Europe Sales
genre_eu = (data.groupby(['Genre'])['EU_Sales'].agg(['mean','sum']))
genre_eu = genre_eu.sort_values(by='sum',ascending=False)
genre_eu.head(10)

mean     sum
Genre                         
Action        0.237902  387.78
Shooter       0.302222  261.12
Sports        0.262654  247.42
Racing        0.283408  164.66
Misc          0.313698  120.46
Role-Playing  0.167416  119.20
Platform      0.269380  108.56
Simulation    0.226566   67.29
Fighting      0.160053   60.50
Strategy      0.094419   25.21

In [41]:
#Genre in Japan Sales
genre_jp = (data.groupby(['Genre'])['JP_Sales'].agg(['mean','sum']))
genre_jp = genre_jp.sort_values(by='sum',ascending=False)
genre_jp.head(10)

mean     sum
Genre                         
Role-Playing  0.172008  122.47
Action        0.046730   76.17
Platform      0.106923   43.09
Sports        0.036667   34.54
Misc          0.086510   33.22
Fighting      0.073677   27.85
Racing        0.047780   27.76
Simulation    0.088081   26.16
Shooter       0.021493   18.57
Puzzle        0.126949   14.98

**For first time Action games in Japan have second place**

### 4.4 Scores Comparations

## 5.Share

Creación de visualizaciones y paneles en Tableau
Abordaje de problemas de accesibilidad al comunicar los datos
Entender el propósito de diferentes herramientas de comunicación empresarial
Narrar una historia basada en datos
Realizar una presentación relacionada con los datos a terceros
Responder preguntas sobre datos

## 6.Act

Codificación en R
Escribir funciones en R
Acceder a datos en R
Limpiar datos en R
Crear visualizaciones de datos en R
Informar sobre análisis de datos a los interesados